examples/neural_networks/plot_mnist_filters.py

```diff
@@ -0,0 +1,53 @@
+"""
+=====================================
+Visualization of MLP weights on MNIST
+=====================================
+
+Sometimes looking at the learned coefficients of a neural network can provide
+inside into the learning behavior. For example if weights look unstructured,
+maybe a weight was not used at all, or if very large coefficients exist, maybe
+regularization was too low or the learning rate too high.
+
+This example shows how to plot some of the first layer weights in a
+MLPClassifier trained on the MNIST dataset.
+
+The input data consists of 28x28 pixel handwritten digits, leading to 784
+features in the dataset. Therefore the first layer weight have the shape (784,
+hidden_layer_sizes[0]).  We can therefore visualize a single column of the
+weight matrix as a 28x28 pixel image.
+
+To make the example run faster, we use very few hidden units, and train only
+for a very short time. Training longer would result in much smoother weights.
+"""
+print(__doc__)
+
+import matplotlib.pyplot as plt
+from sklearn.datasets import fetch_mldata
+from sklearn.neural_network import MLPClassifier
+
+mnist = fetch_mldata("MNIST original")
+# rescale the data, use the traditional train/test split
+X, y = mnist.data / 255., mnist.target
+X_train, X_test = X[:60000], X[60000:]
+y_train, y_test = y[:60000], y[60000:]
+
+#mlp = MLPClassifier(hidden_layer_sizes=(100, 100), max_iter=400, alpha=1e-4,
+#                    algorithm='sgd', verbose=10, tol=1e-4, random_state=1)
+mlp = MLPClassifier(hidden_layer_sizes=(50,), max_iter=10, alpha=1e-4,
```

Comment:
It looks like this:
![figure_1](https://cloud.githubusercontent.com/assets/5950453/9826126/fba77002-58a5-11e5-9a9d-09b03df075d3.png)

But probably because of smaller hidden layer size, test score is only 0.965900